segmentation_models_pytorch/encoders/__init__.py

```diff
@@ -68,20 +97,17 @@ def get_encoder(name, in_channels=3, depth=5, weights=None, output_stride=32, **
         raise KeyError("Wrong encoder name `{}`, supported encoders: {}".format(name, list(encoders.keys())))
 
     params = encoders[name]["params"]
-    params.update(depth=depth)
+    if name.startswith("sam-"):
+        params.update(**kwargs)
+        params.update(dict(name=name[4:]))
+        if depth is not None:
+            params.update(depth=depth)
+    else:
+        params.update(depth=depth)
```

Comment:
is it possible to remove this too?

Reply:
`names` can be removed, but `depth` has to be set to None (default 5) to use a pre-trained model as vit-base comes as with `depth=12`.